assets/js/modules/adsense/dashboard/adsense-module-status.js

```diff
@@ -24,131 +24,303 @@ import SvgIcon from 'GoogleUtil/svg-icon';
+				profile: true,
+				ctaLink: moduleURL,
+				ctaLinkText: __( 'Continue', 'google-site-kit' ),
+				continueAction: false,
+				switchLabel: __( 'Let Site Kit place code on your site', 'google-site-kit' ),
+				switchOffMessage: __( 'You can let Site Kit do this later.', 'google-site-kit' ),
+			};
+			break;
+		case 'ads-display-pending':
+			statusHeadline = __( 'We’re getting your site ready for ads', 'google-site-kit' );
+			statusMessage = __(
+				'This usually takes less than a day, but it can sometimes take a bit longer. We’ll let you know when everything’s ready.',
+				'google-site-kit'
+			);
+			ctaLinkText = __( 'Go to your AdSense account to check on your site’s status', 'google-site-kit' );
+			ctaLink = `https://www.google.com/adsense/new/sites?url=${ siteUrl.hostname }&source=site-kit`;
+			break;
+		case 'account-pending-review':
+			statusHeadline = __( 'We’re getting your site ready for ads', 'google-site-kit' );
+			statusMessage = __(
+				'AdSense is reviewing your site. Meanwhile, make sure you’ve completed these steps in AdSense.',
+				'google-site-kit'
+			);
+			ctaLinkText = __( 'Go to your AdSense account to check on your site’s status', 'google-site-kit' );
+			ctaLink = `https://www.google.com/adsense/new/sites?url=${ siteUrl.hostname }&source=site-kit`;
+			incomplete = true;
+			break;
+		case 'account-required-action':
+			required = true;
+			break;
+		case 'disapproved-account':
+			ctaLink = accountURL;
+			ctaLinkText = __( 'Go to AdSense to find out how to fix the issue', 'google-site-kit' );
+			statusHeadline = __( 'Your site isn’t ready to show ads yet', 'google-site-kit' );
+			statusMessage = __( 'You need to fix some things before we can connect Site Kit to your AdSense account.', 'google-site-kit' );
+			break;
+		case 'disapproved-account-afc':
+			issue = __( 'There is an AdSense account, but the AFC account is disapproved', 'google-site-kit' );
+			icon = 'error';
+			statusHeadline = __( 'Create Account', 'google-site-kit' );
+			statusMessage = __( 'Create an AdMob account, then open AdSense and try to upgrade.', 'google-site-kit' );
+			ctaLinkText = __( 'Create an AdMob Account', 'google-site-kit' );
+			ctaLink = 'https://google.com/admob';
+			break;
+		case 'no-account':
+			statusHeadline = __( 'Create your AdSense account', 'google-site-kit' );
+			statusMessage = __( 'Site Kit will place AdSense code on every page across your site. This means Google will automatically place ads for you in all the best places.', 'google-site-kit' );
+			profile = true;
+			ctaLinkText = __( 'Create AdSense Account', 'google-site-kit' );
+			ctaLink = signupURL;
+			footerText = __( 'Already have an AdSense account?', 'google-site-kit' );
+			footerAppendedText = __( 'to connect to it', 'google-site-kit' );
```

Comment:
This is unrelated to the refactoring here, but this is improper use of translation strings as these should never be concatenated/appended.

I'll create a new issue for this.

Reply:
Hah, good catch. There are so many strings and things happening in this PR that I lost sight of some of the smaller things. Cheers 👍